### DISCUSSION TITLE
[macOS] High input latency unless frame limiter is set to unlimited

### DISCUSSION THREAD

**Discussion by ThatStella7922** [ppy/osu#27767] [Q&A] [answered]
I know what this sounds like, but here's the setup:
Input device: Intuos BT M via USB (056a:0378) using osu!'s OpenTabletDriver
OS: macOS Sonoma 14.4
SoC: M1 Max
Fullscreen mode: Borderless
Renderer: **Metal**
Threading: Multithreaded
Display: 3456x2234, 120 Hz

I'm seeing high input latency with my tablet if I have the frame limiter set to V-Sync, comparable to osu!stable with V-Sync enabled. However, if I set the limiter to unlimited, input latency is almost as low as it is on my desktop PC running osu!stable at Optimal frame limiting, which is what I want.

Problem with that is that osu!lazer starts tearing and having inconsistent frametimes once I disable V-Sync, even though with the Metal renderer and CVDisplayLink synchronization it's still being forced to 120FPS.

Is there some way to get around this, or do I just have to pick between smooth frametimes and low latency input?

**Comment by Joehuu** [comment]
Probably related to https://github.com/ppy/osu-framework/issues/6217.

**Reply by peppy** [reply, marked as answer]
For now you have to pick between them. You can find a middle ground with something like 2x frame limiter, though.

**Reply by ThatStella7922** [reply]
fair enough, hopefully those upcoming CADisplayLink changes help
